### ISSUE TITLE
Device puts itself back into original device group?

### ISSUE THREAD

**Issue by takuy** [Ylianst/MeshCentral#4187]
**Describe the bug**
I'm sure if this is a problem with the agent or Mesh itself, but I originally installed an agent from one Device Group on several workstations. I changed the Device Group afterwards for several devices.

Randomly, the device will change to the original device group. 

Not sure what info or logs I can provide, or if I'm just making bad assumptions here. 

**To Reproduce**
1. Install device group's agent on workstation
2. Move to new device group
3. wait a few days!
4. device moves to different device group?

**Expected behavior**
Devices stays in the assigned device group. 

**Screenshots**
![2022-06-27_22-17-28](https://user-images.githubusercontent.com/1631095/176079296-5a509d63-441c-498b-b0f4-ff440e96781e.png)

![image](https://user-images.githubusercontent.com/1631095/176077569-4673642f-7db5-46e4-a7bd-b7ae240f7ac3.png)

**Server Software (please complete the following information):**
 - OS: RHEL 7
 - Virtualization: Docker
 - Network: WAN, Nginx reverse proxy
 - Version: 1.0.39
 - Node: 12.22.5 
 - Browser: Chrome

**Remote Device (please complete the following information):**
 - Device: Desktop
 - OS: Windows 10 
 - Version: Windows 10 LTSC (1809)
 - Current Core Version (if known): Mar 25 2022, 2265086444

**Comment by silversword411** [comment]
Does it only happen with android devices?

Anything in the log right above that helps give other clues?

**Comment by takuy** [comment]
> Does it only happen with android devices?
> 
> Anything in the log right above that helps give other clues?

My Mesh instance only has Windows devices. Not using in production yet aside from a few production machines for testing. The logs that I can see in the UI haven't been particular helpful. I don't see anything of note other what I've provided. Let me know if there's somewhere else I can check though and I can provide it.

**Comment by takuy** [comment]
> do you maybe have 2 installs of the meshagent running on the test computers?

Only 1 install. Happens on 3 separate workstations. 

> what files are where the meshagent is running? (normally C:\Program Files\MeshAgent\MeshAgent)
> 
> whats the content of the files where meshagent is running? (normally C:\Program Files\MeshAgent\MeshAgent)

![image](https://user-images.githubusercontent.com/1631095/176202010-e78438d4-124f-49df-b32a-c8a6670d4dd1.png)

> can you maybe try uninstalling the agent from a few computers and see if they re-appear in a few days time in the old group?

Are you suggesting to uninstall the agent entirely? And not reinstall?

**Comment by si458** [comment]
whats the content of `MeshAgent.log` ?

also yes you could try uninstalling the agent entirely, and then wait and see if the device re-appears in another group, 
if it does this would mean its running another mesh agent somewhere else?

**Comment by takuy** [comment]
> whats the content of `MeshAgent.log` ?
> 
> also yes you could try uninstalling the agent entirely, and then wait and see if the device re-appears in another group, if it does this would mean its running another mesh agent somewhere else?


```
[2022-05-18 10:32:18 AM] [E00D0B58B943DA56] ..\microstack\ILibParsers.c:10805 (0,0) SelfUpdate -> Current Version: 117444c751eb264f5ebc490db67419387810f6f3
[2022-05-18 10:32:18 AM] [E00D0B58B943DA56] ..\microstack\ILibParsers.c:10805 (0,0) SelfUpdate -> Updating and restarting service...
```

**Comment by silversword411** [comment]
> My Mesh instance only has Windows devices.

Based on those device icons...those look like mobile devices, not windows/pc

![2022-06-28_130106 - mesh icons](https://user-images.githubusercontent.com/13395348/176240785-1c4dc461-9429-4229-bb78-b8e220fbcf25.png)

**Comment by takuy** [comment]
> > My Mesh instance only has Windows devices.
> 
> Based on those device icons...those look like mobile devices, not windows/pc
> 
> ![2022-06-28_130106 - mesh icons]

Right... that's the weird thing I'm seeing in the logs. Not only is it changing device groups, it's changing the "os desc":

`12:00:04 AM - Changed device [computer name] from group Digital Signage: os desc`

I'm just super confused. It seems to always change the "os desc" first and then later move into the default group. Are there logs on the client/agent that would determine what's going on? A verbose mode I can enable?

**Comment by si458** [comment]
you could try setting `"agentLogDump": true` in `config.json`
this Automaticallys downloads all agent error logs into meshcentral-data/agenterrorlogs.txt

**Comment by takuy** [comment]
> 

That's excellent, I'll try that and will report back in a few days.

**Comment by takuy** [comment]
Here's the log:
[agenterrorlogs.txt](https://github.com/Ylianst/MeshCentral/files/9050428/agenterrorlogs.txt)
